internal/consensus/event_publisher.go

```diff
@@ -0,0 +1,98 @@
+package consensus
+
+import (
+	cstypes "github.com/tendermint/tendermint/internal/consensus/types"
+	"github.com/tendermint/tendermint/internal/eventbus"
+	tmevents "github.com/tendermint/tendermint/libs/events"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/types"
+)
+
+// EventPublisher ...
+type EventPublisher struct {
+	// synchronous pubsub between consensus state and reactor.
+	// state only emits EventNewRoundStep, EventValidBlock, and EventVote
+	evsw tmevents.EventSwitch
```

Comment:
there is a difference between event-bus and event-switch implementation
* event-bus uses concurrency
* event-switch uses synchronous execution
also, event-switch is more suitable for tests, because we can pass any data as an event payload
at the moment we are only using event-switch in one test, but we can use it much looser in my opinion
especially, where we need to check updated RoundState